lib/registrant.rb

```diff
@@ -0,0 +1,20 @@
+class Registrant
+    attr_accessor :name,
+                  :age,
+                  :permit,
+                  :license_data
+    def initialize(name, age, permit = false)
+        @name = name
+        @age = age
+        @permit = permit
+        @license_data = {:written=>false, :license=>false, :renewed=>false}      
+    end
+
+    def permit?
+        @permit
+    end
+
+    def earn_permit
+      @permit = true
+    end
+end
```

Comment:
👍 This looks good, but I would change a few things about this class: 

1. We only want to use `attr_accessor`s if we want that attribute to be able to be changed by another class. Since you don't need to do that, you could set all of these as `attr_reader`s which would only allow these attributes to be read outside of this class, but not changed.
2. There is no need for a `permit` method. You created one by listing the `:permit` as an attr_accessor. If you look in the interaction pattern, we never need to call `.permit` on a `Registrant` instance, so for that reason we don't need it listed as an attr_accessor. BUT, we'd still want to keep it as an attribute in our initialize method since our `permit?` method relies on that state.